### ISSUE TITLE
`arrow-parens` should have an option to require single-arg arrow functions to *never* have parentheses

### ISSUE THREAD

**Issue by isiahmeadows** [palantir/tslint#1860]
Some of us like the slightly less verbose version that doesn't include parentheses for things like `map`/`filter`/etc. (i.e. highly functional code). Parentheses can get hard to read in TypeScript's relatively complex grammar, so it would be nice to require them to not exist. (Plus, I encountered this while migrating a JavaScript code base linted with ESLint to TypeScript.)

**Comment by adidahiya** [comment]
If you want to omit parentheses for single-arg arrow functions, I think you should simply turn this rule off. The TypeScript parser already requires you to have parens in multi-arg arrow functions, so I don't see what the point of this option would be.

This code is not valid in the [TS playground](https://www.typescriptlang.org/play/#src=declare%20var%20fn%3A%20(lambda%3A%20(arg1%3A%20boolean%2C%20arg2%3A%20string)%20%3D%3E%20void)%20%3D%3E%20void%3B%0D%0A%0D%0Afn(arg1%2C%20arg2%20%3D%3E%20undefined)%3B%0D%0A):

```ts
declare var fn: (lambda: (arg1: boolean, arg2: string) => void) => void;

fn(arg1, arg2 => undefined);
```

**Comment by isiahmeadows** [comment]
I'm asking because ESLint offers the option to mandate parentheses to be
omitted when they aren't required (as in, single unannotated parameter).
I'm not asking simply for the ability to accept either, but to require
omitting the parentheses. Disabling the rule isn't sufficient.

What I'm requesting is an option to require this:

```ts
// Okay
list.filter(x => x > 0)
list.filter((x: number) => x > 0)

// Error
list.filter((x) => x > 0)
```

Sorry if I'm wording it poorly.

**Comment by adidahiya** [comment]
Ah, cool, I understand now. You're looking for something like a `ban-single-arg-parens` option on the rule that disallows parens when there is only one arg in the lambda. Makes sense; we'd take a PR for this.
